java-checks/src/main/java/org/sonar/java/checks/HiddenFieldCheck.java

```diff
@@ -123,6 +124,18 @@ private void isVariableHidingField(VariableTree variableTree) {
     }
   }
 
+  private static boolean isInStaticInnerClass(VariableTree hiddenVariable, VariableTree variableTree) {
+    Symbol hiddenVariableOwner = hiddenVariable.symbol().owner();
+    Symbol owner = variableTree.symbol().owner();
+    while (owner != null && owner !=hiddenVariableOwner) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)